File: php-sdk/src/Models/CreateUserResponse.php
<?php

declare(strict_types=1);

namespace Cleanster\Models;

/** Response returned when registering a new user account. */
final class CreateUserResponse
{
    public readonly int     $userId;
    public readonly ?string $accessToken;
    /** The original raw array returned by the API. */
    public readonly array   $raw;

    public function __construct(array $data)
    {
        $this->userId      = (int)($data['userId'] ?? 0);
        $this->accessToken = isset($data['accessToken']) && $data['accessToken'] !== null
                             ? (string)$data['accessToken'] : null;
        $this->raw         = $data;
    }
}

File: csharp-sdk/src/Cleanster/Models/CreateUserResponse.cs
using System.Text.Json.Serialization;

namespace Cleanster.Models;

/// <summary>Response returned when registering a new user account.</summary>
public sealed record CreateUserResponse
{
    [JsonPropertyName("userId")]      public int     UserId      { get; init; }
    [JsonPropertyName("accessToken")] public string? AccessToken { get; init; }
}
